### PR TITLE
Test tic-tac-toe public methods

### PR DESCRIPTION
**Pull request by ammarkandel**
Test the public methods using RSpec to test the win cases :
- check the result of the win method. If it true 
- check the result of the win method. If it false
- check if no player wins.
- check if the board is empty.
- check all win cases if it happens or not.
- This check helps to test the winning method that determines the winner.

### CODE REVIEW

**Review by enaburekhan — COMMENTED** [review]
# Invalid
Friendly reminder: You should make sure that your pull request  looks professional before review.
Please take a step back and read a lesson here https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243642/.
Then apply 3 simple rules from this lesson (it will be easy as you can use a template provided there).
This Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready!

You can reach me on slack on @enaburekhan eric 

Happy coding!!!

**Review by NtwaliHeritier — CHANGES_REQUESTED** [review]
### STATUS: CHANGES REQUESTED ### 
Hello team,
Great work so far, however changes are required.
[REQUIRED] Try adding more test cases.
As a tip, you can add negative test cases.

Do the required changes and ask for another review.

File: spec/check_game_result_spec.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      expect_false
+    end
+
+    it 'check the result of the win method. If no player win' do
+      board = %w[X O X O O X X X O]
+      check_game_result
+      expect_false
+    end
+  end
+end
```
Comment (enaburekhan) [inline spec/check_game_result_spec.rb:273]: Kindly add more tests, most especially negative tests cases:pray:

**Review by enaburekhan — CHANGES_REQUESTED** [review]
# Changes Requested :globe_with_meridians:
**Great job so far :clap:. However, there are some changes we need to make to get this project approved :muscle:.**

- Just as the previous Tse pointed out, your test cases are too few. Try to also look at negative events in the course of playing your game and write tests to cater for such events.

All methods implemented in the Enumerables project are tested ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/all_enumerables.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/testing-your-ruby-code))

All the public methods of the classes implemented in project OOP are tested ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/all_tic_tac_toe.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

Kindly make the required changes using the above guidelines and resubmit for another review.

You can reach me on slack @enaburekhan, if you have any questions.

Happy coding!!! :blush:

**Review by BrianSammit — APPROVED** [review]
APPROVED :clap: :green_circle: 
------------
HELLO TEAM!!! Nice job working on the required changes. :muscle: 
--------------
Good luck with the next project. :four_leaf_clover: 
HAPPY CODING!!
